src/library_browser.js

```diff
@@ -560,17 +560,37 @@ var LibraryBrowser = {
     // opposite of native code: In native APIs the positive scroll direction is to scroll up (away from the user).
     // NOTE: The mouse wheel delta is a decimal number, and can be a fractional value within -1 and 1. If you need to represent
     //       this as an integer, don't simply cast to int, or you may receive scroll events for wheel delta == 0.
+    // NOTE: We convert all units returned by events into steps, i.e. individual wheel notches.
+    //       These conversions are only approximations. Changing browsers, operating systems, or even settings can change the values.
     getMouseWheelDelta: function(event) {
       var delta = 0;
       switch (event.type) {
         case 'DOMMouseScroll':
-          delta = event.detail;
+          // 3 lines make up a step
+          delta = event.detail / 3;
           break;
         case 'mousewheel':
-          delta = event.wheelDelta;
+          // 120 units make up a step
+          delta = event.wheelDelta / 120;
           break;
         case 'wheel':
-          delta = event['deltaY'];
+          delta = event.deltaY
+          switch(event.deltaMode) {
+            case 0:
+              // DOM_DELTA_PIXEL: 100 pixels make up a step
+              delta /= 100;
+              break;
+            case 1:
+              // DOM_DELTA_LINE: 3 lines make up a step
+              delta /= 3;
+              break;
+            case 2:
+              // DOM_DELTA_PAGE: A page makes up 80 steps
```

Comment:
The comments are really good and helpful to illustrate the reasoning for the different scale factors. It would be great to add links here to the various specs that specify these. I.e. "3 lines make up a step" - says who it does? The reason for this is due to the "these functions are only approximations". If the approximation here is not correct, developer may have to "fight against" this code to first unapproximate, and then reapproximate for their own scenario, which is why being really specific about this would be good.

To optimize code size with this switch case, perhaps this could be

```js
#if ASSERTIONS
if (!(event.deltaMode >= 0 && event.deltaMode <= 2)) throw 'unrecognized mouse wheel delta mode: ' + event.deltaMode;
#endif
var deltaModeScaleFactors = [100,3,1/80];
delta = event.deltaY / deltaModeScaleFactors[event.deltaMode];
```

for smaller code size.

Reply:
Yes I agree, it's not ideal to make those statements without backing them up.
The problem is that there is barely any official documentation from the browser vendors on this topic. And browsers handle this differently.
I ended up looking for this information in various stackoverflow posts, other implementations, and my own testing. That is why I put this disclaimer above the function.

I do think that this is an improvement over the current state, especially because developers are not even able "to fight against" the current code either.

> To optimize code size with this switch case, perhaps this could be

I usually prefer readability over code size. Isn't his going to get minimized anyway?

Reply:
> I usually prefer readability over code size. Isn't his going to get minimized anyway?

Minification will not be able to perform the kind of transformation I showed above. Though I guess it's ok either way.